store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/operation/handler/GetAllElementsHandler.java

```diff
@@ -29,14 +29,18 @@
 import uk.gov.gchq.gaffer.store.operation.handler.OutputOperationHandler;
 import uk.gov.gchq.gaffer.user.User;
 
-public class GetAllElementsHandler implements OutputOperationHandler<GetAllElements, CloseableIterable<? extends Element>> {
+public class GetAllElementsHandler implements OutputOperationHandler<GetAllElements, Iterable<? extends Element>> {
+
     @Override
-    public CloseableIterable<? extends Element> doOperation(final GetAllElements operation, final Context context, final Store store)
```

Comment:
Same question, why was this returning CloseableIterable before and is there a return type that is Closeable? Or does this make no difference?

Reply:
Not entirely sure what you mean. Originally returned `CloseableIterable` but this was removed. There is not such an interface in the Java libraries that combines `Iterable` and `Closeable` and as we want to return an iterable of results 'Itearble` was chosen as the return type rather than `Closeable`.

Reply:
I am just wondering if this changes functionality as the result is no longer Closeable?

Reply:
That's what I was trying to get across before. The client using Java would have to use the `try-finally` block with the `CloseableUtil` check than the `try` before the change. But returning `Closeable` would mean you would have to check if the `Object` was an `Iterable` or not.

Reply:
So the only thing this did before was stop you having to use the `CloseableUtil` to close it? No functionality differences?

Reply:
Yeah, that is the only difference. The java user will have to include a `try-finally` block with the `CloseableUtil.close()` check if they don't want a memory leak.

Reply:
But they should have done this anyway, so no change really other than name.